### PR TITLE
Fix for regression in multi-output node preview 

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

Fixes:
https://jira.autodesk.com/browse/DYN-3255
https://jira.autodesk.com/browse/DYN-1016 (GH issue: https://github.com/DynamoDS/Dynamo/issues/8885)

Node previews such as that for `List.SortByKey` had regressed after Martin Stacey's port renaming PR's.

Output port names are aligned with the names of the XML return tags specified on the node/method and the objective of Martin's work was to rename the port names by modifying the XML tag names. That was fine. However, for multi-return nodes, the multi-output node preview logic for ZT nodes (which is based on keys of a dictionary returned from the node) also followed the output port names, and these were therefore affected by the XML tag renaming changes, which is why this caused the node preview to break.

The fix is to make the returned dictionary keys to be based on the multi-return attribute inputs and not the output port names (aka XML tags).

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### FYIs
@QilongTang

### CODE REVIEW

File: src/Libraries/CoreNodes/List.cs [hunk 1]
```diff
@@ -484,8 +484,8 @@ public static IDictionary SortByKey(IList list, IList keys)
 
             return new Dictionary<object, object>
             {
-                { "sorted list", sortedList },
-                { "sorted keys", sortedKeys }
+                { "sortedList", sortedList },
+                { "sortedKeys", sortedKeys }
```
Comment (aparajit-pratap) [inline src/Libraries/CoreNodes/List.cs:488]: These name changes are unrelated to the fix and are made only for consistency so that the same is reflected in the preview bubble.

**Comment by mjkkirschner** [comment]
awesome @aparajit-pratap - is there a way to add a regression test for this?

**Comment by aparajit-pratap** [comment]
@mjkkirschner I've added the test, let's wait for PR checks. I also need to update docs for node authors.

**Comment by aparajit-pratap** [comment]
@mjkkirschner any other comments?

File: test/DynamoCoreWpfTests/PreviewBubbleTests.cs [hunk 2]
```diff
@@ -354,6 +355,30 @@ public void PreviewBubble_ShowExpandedPreviewOnPinIconHover()
...
+            Assert.IsTrue(previewBubble.IsCondensed, "Compact preview bubble should be shown");
+            Assert.AreEqual(Visibility.Collapsed, previewBubble.bubbleTools.Visibility, "Pin icon should not be shown");
+
+            // hover preview bubble to see pin icon
+            RaiseMouseEnterOnNode(previewBubble);
+            Assert.AreEqual(Visibility.Visible, previewBubble.bubbleTools.Visibility, "Pin icon should be shown");
+
+            // expand preview bubble
+            RaiseMouseEnterOnNode(previewBubble.bubbleTools);
+            Assert.IsTrue(previewBubble.IsExpanded, "Expanded preview bubble should be shown");
```
Comment (mjkkirschner) [inline test/DynamoCoreWpfTests/PreviewBubbleTests.cs:379]: so this line failed previously?

**Review by mjkkirschner — APPROVED** [review]
LGTM

File: test/DynamoCoreWpfTests/PreviewBubbleTests.cs (hunk 2 above)
Comment (aparajit-pratap) [reply test/DynamoCoreWpfTests/PreviewBubbleTests.cs:379]: Yes
